controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook_test.go

```diff
@@ -161,6 +165,11 @@ func TestKubeadmControlPlaneValidateCreate(t *testing.T) {
 			expectErr: true,
 			kcp:       invalidMaxSurge,
 		},
+		{
+			name:      "should not return an error when maxSurge value is updated to 0",
```

Comment:
Might be I'm missing something, but why are we testing this use case in `TestKubeadmControlPlaneValidateCreate` when the change is in `ValidateUpdate`?

Reply:
This was a total mistake and is fixed.